lib/ruby_smb/gss/provider/ntlm.rb

```diff
@@ -229,25 +236,31 @@ def process_gss_type3(gss_api)
                 domain: type3_msg.domain
               )
               if account.nil?
-                if @provider.allow_anonymous
+                if type3_msg.user == ''
+                  is_guest = false
                   identity = IDENTITY_ANONYMOUS
+                else
+                  is_guest = true
+                  identity = Account.new(type3_msg.user.encode(''.encoding), '', type3_msg.domain.encode(''.encoding)).to_s
```

Comment:
If the user wasn't specified, the identity is anonymous. If the user was specified and the account doesn't exist, it's a guest so use `Account#to_s` to format it properly.